packages/components/src/notice/index.tsx

```diff
@@ -10,24 +10,22 @@ import { __ } from '@wordpress/i18n';
 import { RawHTML, useEffect, renderToString } from '@wordpress/element';
 import { speak } from '@wordpress/a11y';
 import { close } from '@wordpress/icons';
+import type { WPElement } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import { Button } from '../';
-
-/** @typedef {import('@wordpress/element').WPElement} WPElement */
+import Button from '../button';
+import type { NoticeAction, NoticeProps } from './types';
+import type { WordPressComponentProps } from '../ui/context';
+import type { SyntheticEvent } from 'react';
 
 const noop = () => {};
 
-/**
- * Custom hook which announces the message with the given politeness, if a
- * valid message is provided.
- *
- * @param {string|WPElement}     [message]  Message to announce.
- * @param {'polite'|'assertive'} politeness Politeness to announce.
- */
-function useSpokenMessage( message, politeness ) {
+function useSpokenMessage(
+	message: NoticeProps[ 'spokenMessage' ] | WPElement,
```

Comment:
Huh, I always thought `WPElement` was a fake type that was only used in READMEs! I didn't realize it was an aliased export for `ReactElement`. In that case though, `NoticeProps[ 'spokenMessage' ]` (i.e. `ReactNode`) contains `ReactElement` so we don't need it.

```suggestion
	message: NoticeProps[ 'spokenMessage' ],
```

Reply:
Nice, thanks!